### PR TITLE
SDL-0274: add preferred FPS to VideoStreamingCapability

### LINKED ISSUE #3243: [SDL 0274] Add preferred FPS to VideoStreamingCapability
Proposal: [Add preferred FPS to VideoStreamingCapability](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0274-add-preferred-FPS.md)

> This proposal extends SDL video streaming feature by adding preferred FPS to `VideoStreamingCapability`, so that frame rate of video streaming can be adjusted per OEM.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/913

Steering Committee Decision:

> The Steering Committee voted to accept this proposal with revisions.  The revisions will include the items listed in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/913#issuecomment-579062117) from the author.

> The proposal .md file was updated to reflect these revisions on 2/10/2020.

### PR DESCRIPTION
**Pull request by shiniwat**
Fixes #3243 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
N/A

### Summary
This PR is for SDL-0274: add preferred FPS to VideoStreamingCapability.

##### Breaking Changes
* N/A

##### Enhancements
* N/A

##### Bug Fixes
* This takes care of sdl_core issue #3243.
Please note that rpc_spec submodule must be updated. I will update the submodule as soon as another PR at https://github.com/smartdevicelink/rpc_spec/pull/229 has been merged.

### Tasks Remaining:
- none

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by jordynmackool** [comment]
Hi @shiniwat, this PR should point to the branch of the RPC Spec that contains the changes needed for this feature. 

RPC Spec changes are not merged until the feature has been properly tested with all the required projects and it is confirmed the feature will make it into the slated releases.

Once this PR is updated and tested against the RPC Spec feature branch, please tag me requesting a review.

**Comment by jordynmackool** [comment]
@shiniwat is this ready for Livio review?

**Comment by shiniwat** [comment]
Hi @jordynmackool,
While Travis CI seems to fail with "The job exceeded the maximum time limit for jobs, and has been terminated." error, the build works fine on my devbox (Ubuntu 1804). I believe this is ready for Livio review.

**Review by jacobkeeler — APPROVED** [review]

**Review by jacobkeeler — CHANGES_REQUESTED** [review]
See comments

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,4 @@
 	url = https://github.com/smartdevicelink/jsoncpp.git
 [submodule "tools/rpc_spec"]
 	path = tools/rpc_spec
-	url = https://github.com/smartdevicelink/rpc_spec.git
+	url = https://github.com/shiniwat/rpc_spec.git
```
[the remark this answers is not in the document]
Comment (shiniwat) [reply .gitmodules]: @jacobkeeler, thanks for the review. I have updated the commit hash as well as restoring .gitmodules accordingly.

**Review by jacobkeeler — APPROVED** [review]

**Comment by jacobkeeler** [comment]
@shiniwat Please update the rpc_spec submodule to the latest `develop` to fix merge conflicts

**Pull request merged**